lib/sinatra/base.rb

```diff
@@ -179,7 +175,7 @@ def drop_content_info?
     end
 
     def drop_body?
-      DROP_BODY_RESPONSES.include?(status.to_i)
+      !headers["Content-Type"] || DROP_BODY_RESPONSES.include?(status.to_i)
```

Comment:
I'm trying to avoid a proliferation of settings here. I read the RFC you linked (thanks, learn something new every day!) and I think it's fine to send a body without a content-type. Rack::Protection::XSSHeader will set nosniff by default, so a UA can assume octet-stream; if Rack::Protection::XSSHeader is disabled, a UA can detect the content-type. I'll amend the commit.